src/monodroid/jni/monodroid-glue.c

```diff
@@ -704,55 +723,104 @@ typedef void* (*mono_mkbundle_init_ptr) (void (*)(const MonoBundledAssembly **),
 mono_mkbundle_init_ptr mono_mkbundle_init;
 void (*mono_mkbundle_initialize_mono_api) (const BundleMonoAPI *info);
 
-static void
-setup_bundled_app (const char *libappso)
+static char*
+get_full_dso_path (const char *base_dir, const char *dso_path, mono_bool *needs_free)
 {
-	void *libapp;
+	if (needs_free != NULL)
+		*needs_free = FALSE;
+	if (!dso_path)
+		return NULL;
 
-	libapp = dlopen (libappso, RTLD_LAZY);
+	if (base_dir == NULL || dso_path [0] == MONODROID_PATH_SEPARATOR_CHAR)
+		return (char*)dso_path; // Absolute path, can't do much with it
 
-	if (!libapp) {
-		log_fatal (LOG_BUNDLE, "bundled app initialization error: %s", dlerror ());
-		exit (FATAL_EXIT_CANNOT_LOAD_BUNDLE);
-	}
+	char *full_path = path_combine (base_dir, dso_path);
+	if (*needs_free)
+		*needs_free = TRUE;
+	return full_path;
+}
 
-	mono_mkbundle_initialize_mono_api = dlsym (libapp, "initialize_mono_api");
-	if (!mono_mkbundle_initialize_mono_api)
-		log_error (LOG_BUNDLE, "Missing initialize_mono_api in the application");
+static void*
+load_dso (const char *path, int dl_flags, mono_bool skip_exists_check)
```

Comment:
Again, as per previous digression, `load_dso()` implies -- *to me* -- that i'll load "from anywhere," while this only supports looking at filesystem locations.  In keeping with the above, perhaps `load_external_dso()`, or `load_file_dso()`?

Reply:
And my previous reply applies here as well :)